### PR TITLE
Enable ruff annotations ANN206

### PR DESCRIPTION
**Pull request by benesgarage**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/strawberry-graphql/strawberry/issues/2477

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Enable Ruff annotation check ANN206.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Kai Benevento for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: minor
+
+Enable Ruff annotation check ANN206.
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
Release type: patch

This release adds missing annotations in class methods, improving
our type coverage.
```

**Review by patrick91 — APPROVED** [review]
This looks great! thank you so much!

File: RELEASE.md (hunk 1 above)
Comment (patrick91) [reply RELEASE.md:3]: I think we can categorize this as a patch since it doesn't bring new features 😊

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
+
+Enable Ruff annotation check ANN206.
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This release adds missing annotations in class methods, improving
our type coverage.
```

Let's also update this to make it easier for our users to understand what changed 😊

Comment (benesgarage) [reply RELEASE.md]: Sorted, thank you!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**